rust/agama-dbus-server/src/locale.rs

```diff
@@ -89,6 +90,40 @@ impl Locale {
         Ok(())
     }
 
+
+    #[dbus_interface(property, name="UILocale")]
+    fn ui_locale(&self) -> &str {
+        &self.ui_locale
+    }
+
+    #[dbus_interface(property, name="UILocale")]
+    fn set_ui_locale(&mut self, locale: &str) {
+         self.ui_locale = locale.to_string();
+    }
+
+    /// Gets list of locales available on system.
+    /// 
+    /// # Examples
+    /// 
+    /// ```
+    ///   use agama_dbus_server::locale::Locale;
+    ///   let locale = Locale::new();
+    ///   assert!(locale.list_ui_locales().unwrap().len() > 0);
+    /// ```
+    #[dbus_interface(name = "ListUILocales")]
+    pub fn list_ui_locales(&self) -> Result<Vec<String>, Error> {
+        // english is always available ui localization
+        let mut result = vec!["en".to_string()];
+        const DIR : &str = "/usr/share/YaST2/locale/";
+        let entries = read_dir(DIR).context("Reading YaST2 locale")?;
+        for entry in entries {
+            let entry = entry.context("Failed to read entry in YaST2 locale dir")?;
```

Comment:
So we are returning an error if we cannot read any of the dirs, right?

Reply:
yeah, any read failure result in error. I would not expect it unless there is some issue with fs.

Reply:
The `/usr/share/YaST2/locale/` directory is owned by the base `yast2` package so the error should not happen normally.